table/block.cc

```diff
@@ -185,6 +186,91 @@ void DataBlockIter::Seek(const Slice& target) {
   }
 }
 
+void DataBlockIter::SeekForGet(const Slice& target, bool* hash_effective,
+                               bool* found) {
+  assert(data_block_hash_index_);
```

Comment:
I don't see a condition to call SeekForGet only when data_block_hash_index is used. So I think you might trigger this assert in the normal case (unless you have an if-check before calling SeekForGet).